doc/nrf/ug_creating_matter_accessory.rst

```diff
@@ -0,0 +1,274 @@
+.. code-block:: 
+        
+        $MATTER_ROOT/scripts/tools/zap/generate.py src/template.zap
+       
+At this point, new clusters have been added to your Matter accessory. 
+
+Main application
+****************
+
+| Adding clusters alone is not enough to properly model sensor's behaviour.
+| Next step in creating a sensor is changing a way in which an application will interact with newly added clusters. 
+| The easiest way to do it is to modify :file:`src/app_task.cpp`, which contains main loop of the application.
+
+
+Event queue
+===========
+
+| Main application loop is based on an queue on which events are posted by ZCL callbacks, application itself or other entities, like Zephyr timers.
+| In each iteration an event is dequeued and a corresponding event handler is called.
+
+Events
+------
+
+| In a template sample application events are representing:
+
+#. Pressing Button 1.
+#. Releasing Button 1.
+#. Factory reset timeout.
+
+To model behaviour of a sensor we need to add three new events:
+
+#. Sensor activation.
+#. Sensor deactivation.
+#. Sensor measurement.
+
+| Add new events to ``EventType`` in :file:`src/app_event.h`, for example:
+.. code-block:: C++
+        
+        enum EventType : uint8_t { FunctionPress, FunctionRelease, FunctionTimer, SensorActivate, SensorDeactivate, SensorMeasure};
+
+Timers
+------
+
+| One way to make sure that sensor is making measurements in desired time points is to use a Zephyr timer to post ``SensorMeasure`` events.
+| In Template sample such a timer is being used in order to countdown 6 seconds of **Button 1** being pressed to initiate factory reset.
+
+New timer for a measurement event can be added in a similar way:
+
+.. code-block:: C++
+
+        k_timer sSensorTimer;        
+
+        void SensorTimerHandler(k_timer *timer)
+        {
+                GetAppTask().PostEvent(AppEvent{ AppEvent::SensorMeasure });
+        }
+        
+        void StartSensorTimer(uint32_t aTimeoutMs)
+        {
+	        k_timer_start(&sSensorTimer, K_MSEC(aTimeoutMs), K_NO_WAIT);
+        }
+
+        int AppTask::Init()
+        {
+                /*
+                 .      Original content       
+                 */
+
+                k_timer_init(&sSensorTimer, &SensorTimerHandler, nullptr);
+                k_timer_user_data_set(&sSensorTimer, this);
+
+                StartSensorTimer(500);
+                return 0;
+        }
+
+        
+Now every time ``StartSensorTimer(TIMEOUT);`` is called, then after ``TIMEOUT`` milliseconds ``SensorMeasure`` event will be added to the event queue.
+
+
+Event Handlers:
+---------------
+
+| When an event is dequeued the event handler function is called.
+| New events have been added, so corresponding handlers should be implemented, for example (``sensorActive`` will be application variable indicating whether sensor is active):
+.. code-block:: C++
+
+        void AppTask::SensorActivateHandler()
+        {
+	        sensorActive = true;
```

Comment:
I would add this variable declaration on the top of the code and change name to `sSensorActive` to follow the convention of naming global variables.

Reply:
I think that this handler could just start the timer and the one below stop it. No additional variable would be needed then.